src/core/border_router/routing_manager.cpp

```diff
@@ -3327,11 +3332,68 @@ void RoutingManager::RsSender::HandleTimer(void)
 RoutingManager::PdPrefixManager::PdPrefixManager(Instance &aInstance)
     : InstanceLocator(aInstance)
     , mEnabled(false)
+    , mIsRunning(false)
     , mTimer(aInstance)
 {
     mPrefix.Clear();
 }
 
+void RoutingManager::PdPrefixManager::StartStop(bool aStart)
+{
+    Dhcp6PdState oldState = GetState();
+
+    VerifyOrExit(aStart != mIsRunning);
+    mIsRunning = aStart;
+    EvaluateStateChange(oldState);
+
+exit:
+    return;
+}
+
+RoutingManager::Dhcp6PdState RoutingManager::PdPrefixManager::GetState(void) const
+{
+    Dhcp6PdState state = kDhcp6PdStateDisabled;
+
+    if (mEnabled)
+    {
+        if (mIsRunning)
+        {
+            state = kDhcp6PdStateRunning;
+        }
+        else
+        {
+            state = kDhcp6PdStateStopped;
+        }
+    }
+
+    return state;
+}
+
+void RoutingManager::PdPrefixManager::EvaluateStateChange(Dhcp6PdState aOldState)
+{
+    Dhcp6PdState newState = GetState();
+
+    VerifyOrExit(aOldState != newState);
+    switch (newState)
+    {
+    case kDhcp6PdStateDisabled:
+        LogInfo("PdPrefixManager: Disabled");
```

Comment:
Implement a `Dhcp6PdState::ToString()` method and consolidate to a single log line?